### ISSUE TITLE
Confusing and broken example for `--tag-specifications` options in `aws ec2` commands

### ISSUE THREAD

**Issue by thiagomarafeli** [aws/aws-cli#8211]
# TL;DR
- `aws ec2 create-snapshots` has a bad example using `--tag-specifications` flag
- Other `aws ec2` commands seems to have, including the broken one from command above, mixed ways of using this flag in the examples

# Detailed issue
While trying to use `aws ec2 create-snapshots` command with `--tag-specifications` options, I noticed that the example in the `aws ec2 create-snapshots help`, [AWS CLI v1](https://docs.aws.amazon.com/cli/latest/reference/ec2/create-snapshots.html) and [AWS CLI v2](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/ec2/create-snapshots.html) documentations were bad.

In the "Examples" area, it has this example below:
```
aws ec2 create-snapshots \
    --instance-specification InstanceId=i-1234567890abcdef0
    --tag-specifications ResourceType=snapshot,Tags=[{Key=Name,Value=backup},{Key=costcenter,Value=123}]
```

But running with the `--tag-specifications` from above raises this:
```
2023-09-29 15:52:43,968 - MainThread - botocore.hooks - DEBUG - Event provide-client-params.ec2.CreateSnapshots: calling handler <function base64_decode_input_blobs at 0x7f3f5255c2c0>
2023-09-29 15:52:43,968 - MainThread - botocore.hooks - DEBUG - Event before-parameter-build.ec2.CreateSnapshots: calling handler <bound method ParameterAlias.alias_parameter_in_call of <botocore.handlers.ParameterAlias object at 0x7f3f55d84e10>>
2023-09-29 15:52:43,968 - MainThread - botocore.hooks - DEBUG - Event before-parameter-build.ec2.CreateSnapshots: calling handler <function generate_idempotent_uuid at 0x7f3f55d80360>
2023-09-29 15:52:43,968 - MainThread - awscli.clidriver - DEBUG - Exception caught in main()
Traceback (most recent call last):
 File "awscli/clidriver.py", line 460, in main
 File "awscli/clidriver.py", line 595, in _call_
 File "awscli/clidriver.py", line 798, in _call_
 File "awscli/clidriver.py", line 929, in invoke
 File "awscli/clidriver.py", line 941, in _make_client_call
 File "awscli/botocore/client.py", line 341, in _api_call
 File "awscli/botocore/client.py", line 662, in _make_api_call
 File "awscli/botocore/client.py", line 724, in _convert_to_request_dict
 File "awscli/botocore/validate.py", line 336, in serialize_to_request
botocore.exceptions.ParamValidationError: Parameter validation failed:
Invalid type for parameter TagSpecifications[0].Tags[0], value: Key=Name, type: <class 'str'>, valid types: <class 'dict'>
Invalid type for parameter TagSpecifications[0].Tags[1], value: Key=costcenter, type: <class 'str'>, valid types: <class 'dict'>
Invalid type for parameter TagSpecifications[1].Tags[0], value: Key=Name, type: <class 'str'>, valid types: <class 'dict'>
Invalid type for parameter TagSpecifications[1].Tags[1], value: Value=123, type: <class 'str'>, valid types: <class 'dict'>
Invalid type for parameter TagSpecifications[2].Tags[0], value: Value=backup, type: <class 'str'>, valid types: <class 'dict'>
Invalid type for parameter TagSpecifications[2].Tags[1], value: Key=costcenter, type: <class 'str'>, valid types: <class 'dict'>
Invalid type for parameter TagSpecifications[3].Tags[0], value: Value=backup, type: <class 'str'>, valid types: <class 'dict'>
Invalid type for parameter TagSpecifications[3].Tags[1], value: Value=123, type: <class 'str'>, valid types: <class 'dict'>
```

But single or double quoting it worked for me here (currently using AWS CLI 2.13.15 in Ubuntu 22.04.3). Like this:
```
aws ec2 create-snapshots \
    --instance-specification InstanceId=i-1234567890abcdef0
    --tag-specifications "ResourceType=snapshot,Tags=[{Key=Name,Value=backup},{Key=costcenter,Value=123}]"
```

Or this:
```
aws ec2 create-snapshots \
    --instance-specification InstanceId=i-1234567890abcdef0
    --tag-specifications ResourceType=snapshot,"Tags=[{Key=Name,Value=backup},{Key=costcenter,Value=123}]"
```

And, while I was looking for some this problem in the repository, I found out (by [searching](https://github.com/search?q=repo%3Aaws%2Faws-cli+tag-specifications&type=code) for "tag-specifications" in this repo) that this example has multiple variants in code, having different examples between them, which I will link them here below.

It seems confusing for me... what is the actual (recommended) way to use this flag? Some docs appears to not be updated

### Links

https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-ipam.rst#L10
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-ipam.rst#L17
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-subnet.rst#L8
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-subnet.rst#L44
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-subnet.rst#L88
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-image.rst#L43
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/allocate-hosts.rst#L45
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-snapshot.rst#L30
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-ipam-pool.rst#L15
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-ipam-pool.rst#L27
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-ipam-scope.rst#L10
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-ipam-scope.rst#L17
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-vpc.rst#L121
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-vpc.rst#L127
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-internet-gateway.rst#L6
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-verified-access-group.rst#L7
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-instance-event-window.rst#L8
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-instance-event-window.rst#L44
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-verified-access-instance.rst#L6
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-volume.rst#L103
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-verified-access-trust-provider.rst#L9
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-verified-access-endpoint.rst#L14
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-vpc-endpoint.rst#L38
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/run-instances.rst#L176
https://github.com/aws/aws-cli/blob/623e3b77fda3aa34e25410b53ce8021af859f145/awscli/examples/ec2/create-snapshots.rst#L88

**Comment by aBurmeseDev** [comment]
Hi @thiagomarafeli - thanks for reaching out and sorry to hear you're running into this behavior. 

I attempted to replicate this on `aws-cli/2.13.23` without any quoting around the flag and it worked as expected. I tried to look for any changes between the version you're on and the newer one but couldn't spot any changes. The first thing I would try is to update AWS CLI and if you're still seeing the error, I'd be happy to look into the debug logs if you could provide it. You can get the logs by adding `--debug` to your command and please make sure to redact any sensitive info before you share. Thank you!

```
aws ec2 create-snapshots \
    --instance-specification InstanceId=i-<instanceID>
    --tag-specifications ResourceType=snapshot,Tags=[{Key=Name,Value=backup},{Key=costcenter,Value=123}]
{
    "Snapshots": [
        {
            "Description": "",
            "Tags": [],
            "Encrypted": false,
            "VolumeId": "vol-<volumeID>",
            "State": "pending",
            "VolumeSize": 8,
            "StartTime": "2023-10-04T19:07:23.620000+00:00",
            "Progress": "",
            "OwnerId": "<accountID>",
            "SnapshotId": "snap-<snapID>"
        }
    ]
}
```
